app/listings/call_logs_listing.rb

```diff
@@ -34,13 +38,29 @@ def listing_account
   layout filters: :top
 
   filter :id, render: false
-  filter :direction, select_css_class: 'w10'
-  filter :state, select_css_class: 'w10'
+  filter :direction, values: :direction_values, select_css_class: 'w10'
+  filter :state, values: :all_call_log_states, select_css_class: 'w10'
   filter :address, title: 'Caller ID', render: false
   filter :channel_id, render: false
-  filter channel: :name, title: 'Channel'
+  filter channel: :name, values: :all_channel_names, title: 'Channel'
   filter :project_id, render: false
-  filter project: :name, title: 'Project'
+  filter project: :name, values: :all_project_names, title: 'Project'
+
+  def direction_values
+    %w(outgoing incoming)
+  end
+
+  def all_call_log_states
+    CallLog.order("state").pluck("distinct state").reject(&:nil?)
+  end
+
+  def all_channel_names
+    Channel.order("name").pluck("distinct name").reject(&:nil?)
```

Comment:
Should be scoped to the Channels belonging to the projects returned by `listing_account.readable_project_ids`